src/core/scene/a-scene.js

```diff
@@ -36,774 +34,726 @@ if (isIOS) { require('../../utils/ios-orientationchange-blank-bug'); }
-        this.hasWebXR = isWebXRAvailable;
-        this.isAR = false;
-        this.isScene = true;
-        this.object3D = new THREE.Scene();
-        var self = this;
-        this.object3D.onAfterRender = function (renderer, scene, camera) {
-          // THREE may swap the camera used for the rendering if in VR, so we pass it to tock
-          if (self.isPlaying) { self.tock(self.time, self.delta, camera); }
-        };
-        this.resize = bind(this.resize, this);
-        this.render = bind(this.render, this);
-        this.systems = {};
-        this.systemNames = [];
-        this.time = this.delta = 0;
-
-        this.behaviors = {tick: [], tock: []};
-        this.hasLoaded = false;
-        this.isPlaying = false;
-        this.originalHTML = this.innerHTML;
-
-        // Default components.
-        this.setAttribute('inspector', '');
-        this.setAttribute('keyboard-shortcuts', '');
-        this.setAttribute('screenshot', '');
-        this.setAttribute('vr-mode-ui', '');
-        this.setAttribute('device-orientation-permission-ui', '');
-      }
-    },
 
-    addFullScreenStyles: {
-      value: function () {
-        document.documentElement.classList.add('a-fullscreen');
+class AScene extends AEntity {
+  constructor () {
+    var self;
+    super();
+    self = this;
+    self.clock = new THREE.Clock();
+    self.isIOS = isIOS;
+    self.isMobile = isMobile;
+    self.hasWebXR = isWebXRAvailable;
+    self.isAR = false;
+    self.isScene = true;
+    self.object3D = new THREE.Scene();
+    self.object3D.onAfterRender = function (renderer, scene, camera) {
+      // THREE may swap the camera used for the rendering if in VR, so we pass it to tock
+      if (self.isPlaying) { self.tock(self.time, self.delta, camera); }
+    };
+    self.resize = bind(self.resize, self);
+    self.render = bind(self.render, self);
+    self.systems = {};
+    self.systemNames = [];
+    self.time = self.delta = 0;
+
+    self.behaviors = {tick: [], tock: []};
+    self.hasLoaded = false;
+    self.isPlaying = false;
+    self.originalHTML = self.innerHTML;
+  }
+
+  addFullScreenStyles () {
+    document.documentElement.classList.add('a-fullscreen');
+  }
+
+  removeFullScreenStyles () {
+    document.documentElement.classList.remove('a-fullscreen');
+  }
+
+  connectedCallback () {
+    // Defer if DOM is not ready.
+    if (document.readyState === 'loading') {
+      document.addEventListener('DOMContentLoaded', this.connectedCallback.bind(this));
+      return;
+    }
+
+    this.doConnectedCallback();
+  }
+
+  doConnectedCallback () {
+    var self = this;
+    var embedded = this.hasAttribute('embedded');
+
+    // Default components.
+    this.setAttribute('inspector', '');
+    this.setAttribute('keyboard-shortcuts', '');
+    this.setAttribute('screenshot', '');
+    this.setAttribute('vr-mode-ui', '');
+    this.setAttribute('device-orientation-permission-ui', '');
+    super.connectedCallback();
```

Comment:
Note that in previous code, the super method was called before setting the default components. This super method should be called at the top, an example `super.connectedCallback();` is initializing (currently wrongly) `this.sceneEl`, and the component init may use `this.sceneEl` but it's probably not the case here.